--- ./filedigger/main.cpp.orig	2016-08-02 21:22:59.091278333 +0300
+++ ./filedigger/main.cpp	2016-08-02 21:23:39.856804973 +0300
@@ -21,7 +21,7 @@
 #include "filedigger.h"
 #include "mergedvfs.h"
 
-#include <git2/threads.h>
+#include <git2.h>
 
 #include <KAboutData>
 #include <KLocalizedString>
@@ -69,7 +69,12 @@
 	}
 
 	// This needs to be called first thing, before any other calls to libgit2.
+#if !LIBGIT2_VER_MAJOR && LIBGIT2_VER_MINOR <= 20 && !defined(USE_LIBGIT21_API)
 	git_threads_init();
+#else
+	git_libgit2_init();
+#endif
+	 
 	MergedRepository *lRepository = new MergedRepository(NULL, lParser.positionalArguments().first(),
 	                                                     lParser.value("branch"));
 	if(!lRepository->open()) {
@@ -93,6 +98,10 @@
 	lFileDigger->show();
 	int lRetVal = lApp.exec();
 	delete lRepository;
+#if !LIBGIT2_VER_MAJOR && LIBGIT2_VER_MINOR <= 20 && !defined(USE_LIBGIT21_API)
 	git_threads_shutdown();
+#else
+	git_libgit2_shutdown();
+#endif
 	return lRetVal;
 }
--- ./kioslave/bupslave.cpp.orig	2016-08-02 21:35:48.533382353 +0300
+++ ./kioslave/bupslave.cpp	2016-08-02 21:38:19.746641320 +0300
@@ -13,6 +13,8 @@
 #include <grp.h>
 #include <pwd.h>
 
+#include <git2.h>
+
 class BupSlave : public SlaveBase
 {
 public:
@@ -44,14 +46,22 @@
 {
 	mRepository = NULL;
 	mOpenFile = NULL;
+#if !LIBGIT2_VER_MAJOR && LIBGIT2_VER_MINOR <= 20 && !defined(USE_LIBGIT21_API)
 	git_threads_init();
+#else
+	git_libgit2_init();
+#endif
 }
 
 BupSlave::~BupSlave() {
 	if(mRepository != NULL) {
 		delete mRepository;
 	}
+#if !LIBGIT2_VER_MAJOR && LIBGIT2_VER_MINOR <= 20 && !defined(USE_LIBGIT21_API)
 	git_threads_shutdown();
+#else
+	git_libgit2_shutdown();
+#endif
 }
 
 void BupSlave::close() {
