--- ./cmake/plugin.cmake.orig	2017-05-14 19:18:51.147573722 +0300
+++ ./cmake/plugin.cmake	2017-05-14 19:19:15.482007684 +0300
@@ -8,18 +8,8 @@
 	set(CMAKE_SHARED_LIBRARY_PREFIX "")
 endif (WIN32)
 
-find_package(Qt4 REQUIRED)
-
-set(QT_USE_QTNETWORK YES)
-set(QT_USE_QTXML YES)
-include(${QT_USE_FILE})
-
 add_definitions(-DQT_PLUGIN -DQT_SHARED)
 
-qt4_wrap_cpp(MOC_SOURCES ${MOC_HEADERS})
-qt4_wrap_ui(UI_HEADERS ${UIS})	
-qt4_add_resources(RCC_SOURCES ${RCCS})
-
 if (LOCALIZED_LANGS)
     add_translations(TRANSLATIONS ${PLUGIN_NAME} ${HEADERS} ${SOURCES} ${UIS})
 endif(LOCALIZED_LANGS)
